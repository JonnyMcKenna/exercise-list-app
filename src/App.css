.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

select {
  -webkit-appearance: none;
  -moz-appearance: none;
  background: transparent;
  /* background-image: url("./assets/icon/select@3x.png"); */
  background-repeat: no-repeat;
  background-position-x: 90%;
  background-position-y: 50%;
  background-size: 0.7em;
}

.exercise-data {
  /* float: right; */
  /* margin-right: 50px; */
  width: 50%;
  padding: 40px;
  margin: auto;
}

.content {
  clear: both;
  padding-top: 10px;
}

.tableRow {
  width: 100%;
  display: flex;
  margin: 4px 0px 4px 0px;
  padding: 10px;
  border-radius: 8px;
  box-shadow: 1px 1px 2px 2px #e3e3e3;
}

.thirdColumn {
  text-align: -webkit-right;
}

.femaleImage {
  float: right;
  font-size: 18px;
  font-weight: 500;
}

.firstAndSecondColumn {
  width: 85%;
  margin: auto;
}

.maleName {
  padding-left: 60px;
  font-size: 26px;
}

.exerciseName {
  font-size: 26px;
}

@media only screen and (max-width: 1180px) and (min-width: 420px) {
  .exercise-data {
    width: 90%;
  }
}

@media only screen and (max-width: 420px) {

  .exercise-data {
    width: 90%;
    margin-right: 1px;
  }
}

.maleImage {
  max-width: 15%;
}

.femaleImage {
  max-width: 15%;
}
