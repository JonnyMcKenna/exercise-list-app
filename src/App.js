import { useState, useEffect } from "react";
import "./App.css";
import Table from "./components/Table";
import Toggle from "./components/Toggle";
import * as constants from "./constants";

function App() {
  const [exerciseData, setExerciseData] = useState([]);
  const [gender, setGender] = useState(constants.MALE);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    loadExerciseData();
  }, []);

  const loadExerciseData = async () => {
    const response = await fetch(constants.EXERCISE_API);
    const data = await response.json();
    setExerciseData(data);
    setInterval(setLoading(false), 3000);
  };

  return (
    <div className="App">
      <h1 className="title">Exercise List App</h1>

      <Toggle gender={gender} setGender={setGender} />

      {loading ? (
        <div className="loader">Loading...</div>
      ) : (
        <Table exercises={exerciseData.exercises} gender={gender} />
      )}
    </div>
  );
}

export default App;
