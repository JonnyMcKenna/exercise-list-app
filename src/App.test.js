import React from 'react'
import {render} from '@testing-library/react'
import App from './App'

describe('App', () => {
 
 it('should render when still loading', () => {
    const { asFragment } = render(<App />)
    
    expect(asFragment(<App />)).toMatchSnapshot()
   });

  });