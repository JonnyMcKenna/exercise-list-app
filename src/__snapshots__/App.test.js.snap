// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render when still loading 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1
      class="title"
    >
      Exercise List App
    </h1>
    <label
      class="switch"
    >
      <input
        id="togBtn"
        type="checkbox"
      />
      <div
        class="slider round"
      >
        <span
          class="male"
        >
          Male
        </span>
        <span
          class="female"
        >
          Female
        </span>
      </div>
    </label>
    <div
      class="loader"
    >
      Loading...
    </div>
  </div>
</DocumentFragment>
`;
